fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -369,9 +381,12 @@ public Table getTable(String dbName, String tblName) {
                 dbName = dbName.toLowerCase();
                 tblName = tblName.toLowerCase();
             }
-            Table table = IcebergApiConverter.toIcebergTable(icebergTable, catalogName, dbName, tblName, catalogType.name());
+            Database db = getDb(dbName);
+            IcebergTable table =
+                    IcebergApiConverter.toIcebergTable(icebergTable, catalogName, dbName, tblName, catalogType.name());
             table.setComment(icebergTable.properties().getOrDefault(COMMENT, ""));
             tables.put(identifier, icebergTable);
+            updateTableProperty(db, table);
```

Comment:
you can setForeignKeyConstraints in IcebergApiConverter.toIcebergTable, this can avoid creating an iceberg table and then modifying it

Reply:
It's little bit hard here.  We have to call `getDb` to generate constraints.

```
            Database db = getDb(dbName);
            IcebergTable table =
                    IcebergApiConverter.toIcebergTable(icebergTable, catalogName, dbName, tblName, catalogType.name());
            table.setComment(icebergTable.properties().getOrDefault(COMMENT, ""));
            tables.put(identifier, icebergTable);
            updateTableProperty(db, table);
```